### PR TITLE
docs(samples): added create feature and create entity type samples and tests

### PR DESCRIPTION
**Pull request by nayaknishant**

### CODE REVIEW

File: samples/model-builder/conftest.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2021 Google LLC
+# Copyright 2022 Google LLC
```
Comment (morgandu) [inline samples/model-builder/conftest.py:1]: The year change is only needed when adding a new file.

File: samples/model-builder/conftest.py [hunk 2]
```diff
@@ -364,3 +364,48 @@ def mock_endpoint_explain(mock_endpoint):
...
+def mock_create_featurestore():
+    with patch.object(aiplatform.Featurestore, "create") as mock:
+        yield mock
+
+
+@pytest.fixture
+def mock_create_entity_type():
+    with patch.object(aiplatform.EntityType, "create") as mock:
+        yield mock
+
```
Comment (morgandu) [inline samples/model-builder/conftest.py:399]: you will need `mock_entity_type` and `mock_feature` similar to `mock_featurestore`, which are the return value for `mock_create_*` and `mock_get_*`

File: samples/model-builder/test_constants.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2021 Google LLC
+# Copyright 2022 Google LLC
```
Comment (morgandu) [inline samples/model-builder/test_constants.py:1]: 2021

**Review by morgandu — CHANGES_REQUESTED** [review]

**Comment by morgandu** [comment]
left some comment throughout, please add sync arg for relevant methods, and create a section for featurestore constant.

File: samples/model-builder/conftest.py (hunk 2 above)
Comment (nayaknishant) [reply samples/model-builder/conftest.py:399]: Added functions

File: samples/model-builder/test_constants.py (hunk 3 above)
Comment (nayaknishant) [reply samples/model-builder/test_constants.py:1]: Fixed

File: samples/model-builder/conftest.py (hunk 1 above)
Comment (nayaknishant) [reply samples/model-builder/conftest.py:1]: Fixed

File: samples/model-builder/conftest.py [hunk 4]
```diff
@@ -364,3 +364,47 @@ def mock_endpoint_explain(mock_endpoint):
...
+@pytest.fixture
+def mock_create_featurestore():
+    with patch.object(aiplatform.Featurestore, "create") as mock:
+        yield mock
+
+
+@pytest.fixture
+def mock_create_entity_type():
+    with patch.object(aiplatform.EntityType, "create") as mock:
```
Comment (morgandu) [inline samples/model-builder/conftest.py:397]: Please follow the same pattern for resource creation below
```
@pytest.fixture
def mock_create_featurestore(mock_featurestore):
    with patch.object(
        aiplatform.featurestore.Featurestore, "create"
    ) as mock_create_featurestore:
        mock_create_featurestore.return_value = mock_featurestore
        yield mock_create_featurestore
```

Comment (morgandu) [reply samples/model-builder/conftest.py:397]: this need to apply to `mock_feature` as well.

File: samples/model-builder/create_entity_type_sample.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+):
+
+    aiplatform.init(project=project, location=location)
+
+    fs = aiplatform.EntityType.create(
+        entity_type_id=entity_type_id, featurestore_name=featurestore_name
+    )
+
+    fs.wait()
+
```
Comment (morgandu) [inline samples/model-builder/create_entity_type_sample.py:31]: https://github.com/googleapis/python-aiplatform/blob/main/google/cloud/aiplatform/featurestore/entity_type.py#L470

https://github.com/googleapis/python-aiplatform/blob/main/google/cloud/aiplatform/featurestore/featurestore.py#L517

Please see detail in the return type hint, and choose a parameter name accordingly.

File: samples/model-builder/create_feature_sample.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    featurestore_id: str,
+):
+
+    aiplatform.init(project=project, location=location)
+
+    fs = aiplatform.Feature.create(
+        feature_id=feature_id,
+        value_type=value_type,
+        entity_type_name=entity_type_name,
+        featurestore_id=featurestore_id,
```
Comment (morgandu) [inline samples/model-builder/create_feature_sample.py:35]: When `entity_type_name` is a fully qualified resource name, `featurestore_id`, `project` and `location` will not be needed.

When `entity_type_name` is a resource ID, `featurestore_id`, `project` and `location` will all be needed.

Please see detail in https://github.com/googleapis/python-aiplatform/blob/main/google/cloud/aiplatform/featurestore/feature.py#L507
and correct this sample.

File: samples/model-builder/create_feature_sample.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    )
+
+    fs.wait()
+
+    return fs
```
Comment (morgandu) [inline samples/model-builder/create_feature_sample.py:40]: fs -> f or my_feature

File: samples/model-builder/create_feature_sample_test.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    mock_sdk_init.assert_called_once_with(
+        project=constants.PROJECT, location=constants.LOCATION
+    )
+
+    mock_create_feature.assert_called_once_with(
+        feature_id=constants.FEATURE_ID,
+        value_type=constants.FEATURE_VALUE_TYPE,
+        entity_type_name=constants.ENTITY_TYPE_ID,
+        featurestore_id=constants.FEAUTURESTORE_NAME,
+    )
```
Comment (morgandu) [inline samples/model-builder/create_feature_sample_test.py:39]: correction for sample arg usage and mock assert are both needed

File: samples/model-builder/test_constants.py [hunk 9]
```diff
@@ -197,3 +197,12 @@
 )
 PYTHON_MODULE_NAME = "trainer.task"
 MODEL_TYPE = "CLOUD"
+
+# Feature store constants
+FEAUTURESTORE_NAME = "featurestore_sample"
+ENTITY_TYPE_ID = "entity_type_sample"
```
Comment (morgandu) [inline samples/model-builder/test_constants.py:203]: can have both `*_ID`, and `*_NAME` for entity_type and feature, as recommended in the previous PR.

**Review by morgandu — CHANGES_REQUESTED** [review]

File: samples/model-builder/create_entity_type_sample.py (hunk 5 above)
Comment (nayaknishant) [reply samples/model-builder/create_entity_type_sample.py:31]: Fixed, changed variable name to `my_entity_type`

File: samples/model-builder/create_feature_sample.py (hunk 6 above)
Comment (nayaknishant) [reply samples/model-builder/create_feature_sample.py:35]: Fixed, changed passed argument from `entity_type_name` to `entity_type_id`

File: samples/model-builder/create_feature_sample.py (hunk 7 above)
Comment (nayaknishant) [reply samples/model-builder/create_feature_sample.py:40]: Fixed, changed to `my_feature`

File: samples/model-builder/test_constants.py (hunk 9 above)
Comment (nayaknishant) [reply samples/model-builder/test_constants.py:203]: added new constants

File: samples/model-builder/conftest.py (hunk 4 above)
Comment (nayaknishant) [reply samples/model-builder/conftest.py]: Fixed

**Review by morgandu — APPROVED** [review]

**Comment by morgandu** [comment]
LGTM! Looks Good!

**Pull request merged**